pkg/compact/compact.go

```diff
@@ -206,6 +208,11 @@ type Grouper interface {
 	// Groups returns the compaction groups for all blocks currently known to the syncer.
 	// It creates all groups from the scratch on every call.
 	Groups(blocks map[ulid.ULID]*metadata.Meta) (res []*Group, err error)
+	// Starting a group lets the grouper know that the blocks in the groups are currently
+	// being compacted and to ignore them in future plans
+	StartGroup(group *Group)
```

Comment:
Why not have `Group.Lock` method? 🤗

Reply:
Will update. Thanks for taking a look at these changes. I agree most of the work should be done by the planner, the reason i didn't go that path is the number of changes required would be substantial. Happy to go down that path if it's preferred

Reply:
After a second look Group lock wont work because its the grouper that needs to know about whats currently running.